tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/DataTypes/dataTypes.spec.ts

```diff
@@ -0,0 +1,191 @@
+import {AliasHelper, ConstantHelper, test} from '@umbraco/playwright-testhelpers';
+import {expect} from "@playwright/test";
+import {
+  ApprovedColorPickerDataTypeBuilder,
+  DocumentTypeBuilder,
+  TextBoxDataTypeBuilder
+} from "@umbraco/json-models-builders";
+
+test.describe('DataTypes', () => {
+
+  test.beforeEach(async ({page, umbracoApi}) => {
+    await umbracoApi.login();
+  });
+
+  test('Tests Approved Colors', async ({page, umbracoApi, umbracoUi}) => {
+    const name = 'Approved Colour Test';
+    const alias = AliasHelper.toAlias(name);
+
+    await umbracoApi.documentTypes.ensureNameNotExists(name);
+    await umbracoApi.content.deleteAllContent();
+    await umbracoApi.dataTypes.ensureNameNotExists(name);
+    await umbracoApi.templates.ensureNameNotExists(name);
+
+    const pickerDataType = new ApprovedColorPickerDataTypeBuilder()
+      .withName(name)
+      .withPrevalues(['000000', 'FF0000'])
+      .build()
+    await umbracoApi.content.createDocTypeWithContent(name, alias, pickerDataType);
+
+    // This is an ugly wait, but we have to wait for cache to rebuild
+    await page.waitForTimeout(5000);
+
+    // Editing template with some content
+    await umbracoApi.templates.edit(name,
+      '@inherits Umbraco.Cms.Web.Common.Views.UmbracoViewPage<ApprovedColourTest>' +
+      '\n@{' +
+      '\n    Layout = null;' +
+      '\n}' +
+      '\n<p style="color:@Model.UmbracoTest">Lorem ipsum dolor sit amet</p>');
+
+    // Enter content
+    await umbracoUi.refreshContentTree();
+    await umbracoUi.clickElement(umbracoUi.getTreeItem("content", [name]));
+
+    // Pick a colour
+    await page.locator('.btn-000000').click();
+
+    // Save
+    await umbracoUi.clickElement(umbracoUi.getButtonByLabelKey(ConstantHelper.buttons.saveAndPublish));
+    await umbracoUi.isSuccessNotificationVisible();
+
+    // Assert
+    const expected = `<p style="color:000000" > Lorem ipsum dolor sit amet </p>`;
+    await expect(umbracoApi.content.verifyRenderedContent('/', expected, true)).toBeTruthy();
+    await expect(await page.locator('.umb-button__overlay')).not.toBeVisible();
+
+    // Pick another colour to verify both work
+    await page.waitForTimeout(5000);
+    await page.locator('.btn-FF0000').click();
+
+    // Save
+    await umbracoUi.clickElement(umbracoUi.getButtonByLabelKey(ConstantHelper.buttons.saveAndPublish));
+    await umbracoUi.isSuccessNotificationVisible();
+    await expect(await umbracoUi.getSuccessNotification()).toBeVisible();
+    await expect(await page.locator('.umb-button__overlay')).not.toBeVisible();
+    await page.waitForTimeout(5000);
```

Comment:
Same again, the verifyRenderedContent should not retry on its own, this timeout should not be necesarry